pennylane/compiler/compiler.py

```diff
@@ -0,0 +1,214 @@
+    # by validating that all entry points (qjit, context, and ops) are defined.
+    for _, eps_dict in AvailableCompilers.names_entrypoints.items():
+        ep_interface = AvailableCompilers.entrypoints_interface
+        if any(ep not in eps_dict.keys() for ep in ep_interface):
+            raise KeyError(f"expected {ep_interface}, but recieved {eps_dict}")
+
+
+# Scan installed compiler packages
+# and update AvailableCompilers
+_refresh_compilers()
+
+
+def _reload_compilers():
+    """Reload and scan installed PennyLane compiler packages to refresh the
+    compilers names and entry points.
+    """
+
+    # Note re-importing ``importlib.metadata`` can be a very slow operation
+    # on systems with a large number of installed packages.
+
+    reload(metadata)
+    _refresh_compilers()
+
+
+def available_compilers() -> List[str]:
+    """Loads and returns a list of available compilers that are
+    installed and compatible with the :func:`~.qjit` decorator.
+
+    **Example**
+
+    This method returns the name of installed compiler packages supported in
+    PennyLane. For example, after installing the
+    `Catalyst <https://github.com/pennylaneai/catalyst>`__
+    compiler, this will now appear as an available compiler:
+
+    >>> qml.compiler.available_compilers()
+    ['catalyst']
+    """
+
+    # Reload installed packages and updates
+    # the class variable names_entrypoints
+    _reload_compilers()
+
+    return list(AvailableCompilers.names_entrypoints.keys())
+
+
+def available(compiler="catalyst") -> bool:
+    """Check the availability of the given compiler package.
+
+    Args:
+        compiler (str): name of the compiler package (default value is ``catalyst``)
+
+    Return:
+        bool: ``True`` if the compiler package is installed on the system
+
+    **Example**
+
+    Before installing the ``pennylane-catalyst`` package:
+
+    >>> qml.compiler.available("catalyst")
+    False
+
+    After installing the ``pennylane-catalyst`` package:
+
+    >>> qml.compiler.available("catalyst")
+    True
+    """
+
+    # It only refreshes the compilers names and entry points if
+    # the name is not already stored.
+
+    if compiler not in AvailableCompilers.names_entrypoints:
+        # Reload installed packages and updates
+        # the class variable names_entrypoints
+        _reload_compilers()
+
+    return compiler in AvailableCompilers.names_entrypoints
+
+
+def active_compiler() -> Optional[str]:
+    """Check which compiler is activated inside a :func:`~.qjit` evaluation context.
+
+    This helper function may be used during implementation
+    to allow differing logic for transformations or operations that are
+    just-in-time compiled, versus those that are not.
+
+    Return:
+        Optional[str]: Name of the active compiler inside a :func:`~.qjit`
+            evaluation context. If there is no active compiler, ``None``
+            will be returned.
```

Comment:
```suggestion
        evaluation context. If there is no active compiler, ``None``
        will be returned.
```
Not showing up right:
![image](https://github.com/PennyLaneAI/pennylane/assets/49409390/d0bb3323-2456-4480-93a5-2ea081d7357b)

Reply:
Fixed in PR #4812